### PR TITLE
Seabeta 487 capture google analytics events

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs
Comment (CodrinCotarlanSoftwire) [inline SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs:1145]: I might be able to dig up the client ID if I look into the cookies, but it's not needed

File: SeaPublicWebsite/ExternalServices/GoogleAnalytics/GoogleAnalyticsService.cs
Comment (CodrinCotarlanSoftwire) [inline SeaPublicWebsite/ExternalServices/GoogleAnalytics/GoogleAnalyticsService.cs:45]: Documentation says it's an object. I could make it a `Dictionary<string, object>` and achieve the same result

File: SeaPublicWebsite/Views/Shared/_Layout.cshtml
Comment (CodrinCotarlanSoftwire) [inline SeaPublicWebsite/Views/Shared/_Layout.cshtml:10]: Forgot to remove this at some point

File: SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs
Comment (DanCorderSoftwire) [reply SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs:1145]: What is the client ID used for?

File: Infrastructure/SetEnvironmentSecretsTemplate.sh
Comment (DanCorderSoftwire) [inline Infrastructure/SetEnvironmentSecretsTemplate.sh:32]: We need to run this on the environments before this ticket is merged

File: SeaPublicWebsite/ExternalServices/GoogleAnalytics/GoogleAnalyticsService.cs
Comment (DanCorderSoftwire) [reply SeaPublicWebsite/ExternalServices/GoogleAnalytics/GoogleAnalyticsService.cs:45]: `Dictionary<string, object>` seems nicer if it works

File: SeaPublicWebsite/wwwroot/js/bundle/google-analytics.js
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/wwwroot/js/bundle/google-analytics.js:34]: For text inputs does this event fire only when the control loses focus, or every time the user types a character?

**Review by DanCorderSoftwire — CHANGES_REQUESTED** [review]

File: SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs
Comment (CodrinCotarlanSoftwire) [reply SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs:1145]: So that we know which user caused the event. I will dig up the id from the _ga cookie and use it here instead

File: SeaPublicWebsite/wwwroot/js/bundle/google-analytics.js
Comment (CodrinCotarlanSoftwire) [reply SeaPublicWebsite/wwwroot/js/bundle/google-analytics.js:34]: Yes, that is why I used onChange. The event is only fired when focus is moved away.

File: Infrastructure/SetEnvironmentSecretsTemplate.sh
Comment (CodrinCotarlanSoftwire) [reply Infrastructure/SetEnvironmentSecretsTemplate.sh:32]: I'll let you merge the ticket once the environment is set up

**Review by DanCorderSoftwire — APPROVED** [review]

**Pull request merged**
